lib/pbench/test/unit/server/test_datasets_compare.py

```diff
@@ -0,0 +1,160 @@
+from http import HTTPStatus
+from pathlib import Path
+from typing import Optional
+
+from pquisby.lib.post_processing import QuisbyProcessing
+import pytest
+import requests
+
+from pbench.server import JSON
+from pbench.server.cache_manager import CacheManager, TarballUnpackError
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound, Metadata
+
+
+def mock_get_value(dataset: Dataset, key: str, user: Optional = None) -> str:
+    if dataset.name == "uperf_3" or dataset.name == "uperf_4":
+        return "hammerDB"
+    return "uperf"
+
+
+class TestCompareDatasets:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, get_token_func):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            get_token_func: Pbench token fixture
+        """
+
+        def query_api(
+            datasets: list, user: str, expected_status: HTTPStatus
+        ) -> requests.Response:
+            ds_list = []
+            for dataset in datasets:
+                try:
+                    dataset_id = Dataset.query(name=dataset).resource_id
+                    ds_list.append(dataset_id)
+                except DatasetNotFound:
+                    ds_list.append(dataset)  # Allow passing deliberately bad value
+            headers = None
+            if user:
+                headers = {"authorization": f"bearer {get_token_func(user)}"}
+            response = client.get(
+                f"{server_config.rest_uri}/compare",
+                query_string={"datasets": ds_list},
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def mock_compare_csv_to_json(self, benchmark_name, input_type, data_stream) -> JSON:
+        return {"status": "success", "json_data": "quisby_data"}
```

Comment:
With the change to use parametrization, this function is now used in only one place; consider making it an inner function (or even a lamba, although it's unwieldy) inside `test_datasets_with_different_benchmark()`.